storage/storage_dest.go

```diff
@@ -555,20 +589,25 @@ func (s *storageImageDestination) queueOrCommit(index int, info addedLayerInfo)
 	return nil
 }
 
-// getDiffIDOrTOCDigest returns the diffID for the specified digest or the digest for the TOC, if known.
-func (s *storageImageDestination) getDiffIDOrTOCDigest(uncompressedDigest digest.Digest) (digest.Digest, bool) {
+// getLayerID returns the diffID for the specified digest or the digest for the TOC, if known.
+func (s *storageImageDestination) getLayerID(uncompressedDigest digest.Digest, tocDigest digest.Digest) (digest.Digest, bool) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	if d, found := s.diffOutputs[uncompressedDigest]; found {
-		return d.TOCDigest, found
+	if tocDigest != "" {
+		if d, found := s.tocDigests[tocDigest]; found {
+			return d, found
```

Comment:
Please have `getLayerID` return a string which is not a possible digest, maybe `"TOC:"+d.String()`, to ensure the inputs to the hashing don’t have collisions.